deepchem/models/tensorgraph/tests/test_layers_eager.py

```diff
@@ -0,0 +1,980 @@
+import deepchem as dc
+import numpy as np
+import tensorflow as tf
+import deepchem.models.tensorgraph.layers as layers
+import tensorflow.contrib.eager as tfe
+from tensorflow.python.eager import context
+from tensorflow.python.framework import test_util
+import rdkit
+
+
+class TestLayersEager(test_util.TensorFlowTestCase):
+  """
+  Test that layers function in eager mode.
+  """
+
+  def test_conv_1d(self):
+    """Test invoking Conv1D in eager mode."""
+    with context.eager_mode():
+      with tfe.IsolateTest():
+        width = 5
+        in_channels = 2
+        filters = 3
+        kernel_size = 2
+        batch_size = 10
+        input = np.random.rand(batch_size, width, in_channels).astype(
+            np.float32)
+        layer = layers.Conv1D(filters, kernel_size)
+        result = layer(input)
+        self.assertEqual(result.shape[0], batch_size)
+        self.assertEqual(result.shape[2], filters)
+        assert len(layer.variables) == 2
+
+        # Creating a second layer should produce different results, since it has
```

Comment:
Maybe a dumb question but how would the syntax work if you wanted to reuse the weights? for a new "input"

Reply:
Just invoke it again with a different input: `result2 = layer(input2)`.